src/stores/DialogStore/DialogStore.ts

```diff
@@ -20,179 +62,78 @@ export class DialogStore {
+                    if (props && props.oldLayoutName) {
+                        AppStore.Instance.layoutStore.setOldLayoutName(props.oldLayoutName);
+                    }
+                    this.zIndexManager.assignIndex(DialogId.Layout);
+                    break;
+
+                case DialogId.Workspace:
+                    this.dialogVisible.set(DialogId.FileBrowser, false);
+                    this.workspaceDialogMode = props.mode;
+                    this.zIndexManager.assignIndex(DialogId.Workspace);
+                    break;
+
+                case DialogId.FileBrowser:
+                    this.workspaceDialogMode = WorkspaceDialogMode.Hidden;
+                    this.dialogVisible.set(DialogId.FileBrowser, true);
+                    this.zIndexManager.assignIndex(DialogId.FileBrowser);
+                    break;
+
+                case DialogId.ExistingSnippet:
+                    if (props.snippet) {
+                        SnippetStore.Instance.setActiveSnippet(props.snippet, props.name);
+                    }
+                    if (!this.dialogVisible.get(DialogId.Snippet)) {
+                        this.dialogVisible.set(DialogId.Snippet, true);
+                        this.zIndexManager.assignIndex(DialogId.Snippet);
+                    }
+                    break;
+
+                case DialogId.NewSnippet:
+                    SnippetStore.Instance.clearActiveSnippet();
+                    if (!this.dialogVisible.get(DialogId.Snippet)) {
+                        this.dialogVisible.set(DialogId.Snippet, true);
+                        this.zIndexManager.assignIndex(DialogId.Snippet);
+                    }
+                    break;
+
+                case DialogId.ExternalPage:
+                    this.externalPageDialogUrl = props.url;
+                    this.externalPageDialogTitle = props.title;
+                    this.dialogVisible.set(DialogId.ExternalPage, true);
+                    this.zIndexManager.assignIndex(DialogId.ExternalPage);
+                    break;
+
+                default:
+                    this.dialogVisible.set(id, true);
+                    this.zIndexManager.assignIndex(id);
+                    break;
+            }
+        } else {
+            this.zIndexManager.updateIndexOnSelect(id);
+        }
     };
 
-    // File Browser
-    @observable fileBrowserDialogVisible: boolean = false;
-    @action showFileBrowserDialog = () => {
-        this.workspaceDialogMode = WorkspaceDialogMode.Hidden;
-        this.fileBrowserDialogVisible = true;
-    };
-    @action hideFileBrowserDialog = () => {
-        this.fileBrowserDialogVisible = false;
+    @action hideDialog = (id: string) => {
+        if (id === DialogId.Workspace) {
+            this.workspaceDialogMode = WorkspaceDialogMode.Hidden;
+        } else {
+            this.dialogVisible.set(id, false);
+        }
+        this.zIndexManager.updateIndexOnRemove(id);
+        this.zIndexManager.removeIndex(id);
```

Comment:
I suggest we move `removeIndex` in `updateIndexOnRemove`.

Reply:
okay.

Reply:
Please remove commented `removeIndex`.

Reply:
okay